sklearn/ensemble/voting_classifier.py

```diff
@@ -0,0 +1,202 @@
+        """ 
+        Fits the estimators.
+        
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+            
+        y : array-like, shape = [n_samples]
+            Target values.
+      
+        Returns
+        -------
+        self : object
+
+        """
+        for clf in self.clfs:
+            clf.fit(X, y)
+        return self
+            
+    def predict(self, X):
+        """
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+        
+        Returns
+        ----------
+        maj : array-like, shape = [n_class_labels]
+            Predicted class labels by majority rule.
+        
+        """
+        if self.voting == 'soft':
+            avg = self.predict_proba(X)
+            maj = np.apply_along_axis(lambda x: max(enumerate(x), key=operator.itemgetter(1))[0], axis=1, arr=avg)
+        
+        else: # 'hard' voting
+            self.classes_ = self._predict(X)
+            
+            # arange class labels as [n_class_labels, n_classifier_predictions]
+            self.classes_ = np.asarray([self.classes_[:,c] for c in range(self.classes_.shape[1])])
+            
+            # duplicate class labels if weights are provided for argmax majority rule
+            if self.weights:
+                self.classes_ = np.concatenate([np.tile(self.classes_[:,c,None], w)
+                                        for w,c in zip(self.weights, range(self.classes_.shape[1]))], axis=1)
+
+            maj = np.apply_along_axis(lambda x: np.argmax(np.bincount(x)), axis=1, arr=self.classes_)
+        
+        return maj
+            
+    
+    def predict_proba(self, X):
+        """
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+        
+        Returns
+        ----------
+        avg : array-like, shape = [n_samples, n_probabilities]
+            Weighted average probability for each class per sample.
+        
+        """
+        self.probas_ = self._predict_probas(X)
+        avg = np.average(self.probas_, axis=0, weights=self.weights)
+        
+        return avg
+
+  
+    def transform(self, X):
+        """         
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+      
+        Returns
+        -------
+        If not `weights=None`:
```

Comment:
Also, I don't think this is valid numpy doc format ;)
Lastly, I think the you should flatten the array to be ``(n_samples, *)`` and not have ndim=3, so that can directly be used in a pipeline.